proposals/0024-opacity-micromaps.md

```diff
@@ -200,46 +217,79 @@ In addition to the AST traversal detecting any explicit use of the new flag,
 
-Validation will be added to ensure that the shader model is at least 6.9 when
-the `RaytracingPipelineFlags::AllowOpacityMicromaps` is used in a
-[Raytracing pipeline config1][pipeline-config] subobject.
+A check will be added on the declaration of a RayQuery object 
+(when not dependent), so that when `RAY_FLAG_FORCE_OMM_2_STATE` is set on
+the RayFlags template argument, and `RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS`
+is not set on the RayQueryFlags template argument, a DefaultError warning 
+is emitted.
 
-Proposed validation error diagnostic:
+Proposed DefaultError warning diagnostic:
+- `"When using 'RAY_FLAG_FORCE_OMM_2_STATE' in RayFlags, RayQueryFlags must have RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS set."`.
 
-- `"RaytracingPipelineFlags in RaytracingPipelineConfig1 subobject '%0' specifies unknown flags (0x%1) for shader model %2; valid mask: 0x%3"`
+A check will be added at the call site of `RayQuery::TraceRayInline` that will
+emit a DefaultError warning when a constant `RayFlags` parameter has the
+`RAY_FLAG_FORCE_OMM_2_STATE` flag set, and the RayQuery object's
+`RayQueryFlags` template argument does not have the
+`RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS` flag set.
+The same warning message is emitted, but a diagnostic note should also
+point to the RayQuery object declaration, where this RayQueryFlag needs to be
+specified.
 
+#### Validation Changes
 Three DXIL operations accept `RayFlags` as input, but only one requires this
 input to be immediate: `AllocateRayQuery`.
 
-Validation will be added to check the `RayFlags` parameters for each applicable
-DXIL operation, with an error emitted if the flags are constant, the new flag
-is used, and the shader model is less than 6.9.
+Validation will be added to check the `RayFlags` parameters, and, if 
+applicable, the `RayQueryFlags` parameters, for each applicable DXIL operation,
+including the new `AllocateRayQuery2` operation. An error will be emitted if 
+the flags are constant, the new flag is used, and the shader model 
+is less than 6.9.
 
 Proposed validation error diagnostic:
 
-- `"RayFlags used in '%0' specifies unknown flags (0x%1) for shader model %2; valid mask: 0x%3"`
+- `"RayFlags used in '%0' specifies unknown flags (0x%1) for shader model %2"`
 
 Validation will also be added to ensure the flags are constant on input to
-the `AllocateRayQuery` DXIL operation.
+the `AllocateRayQuery` and `AllocateRayQuery2` DXIL operation.
 
-Proposed validation error diagnostic:
+Proposed validation error diagnostics:
 
 - `"ConstRayFlags argument of AllocateRayQuery '%0' must be constant"`
+- `"ConstRayFlags and RayQueryFlags arguments of AllocateRayQuery2 '%0' must be constant"`
+
+Finally, validation will be added for the `AllocateRayQuery2` DXIL operation
+to ensure that when `RAY_FLAG_FORCE_OMM_2_STATE` is set on the ConstRayFlags
+argument, the `RAYQUERY_FLAG_ALLOW_OPACITY_MICROMAPS` is also set on the
+RayQueryFlags argument.
```

Comment:
Just a reminder to synchronize argument names referred to here with whatever's used in the signature above, and use the same for the hctdb.py argument names.  It turns out that `ConstRayFlags` should be `constRayFlags`, as it's currently defined for `AllocateRayQuery`.